pkg/controllers/user/alert/watcher/statefulset.go

```diff
@@ -67,39 +67,49 @@ func (w *StatefulsetWatcher) watchRule() error {
 		if alert.Status.AlertState == "inactive" {
 			continue
 		}
-		if alert.Spec.TargetWorkload.Type == "statefulset" {
-			if alert.Spec.TargetWorkload.WorkloadID != "" {
-				parts := strings.Split(alert.Spec.TargetWorkload.WorkloadID, ":")
-				ns := parts[0]
-				id := parts[1]
-				ss, err := w.statefulsetLister.Get(ns, id)
-				if err != nil {
-					logrus.Debugf("Failed to get statefulset %s: %v", id, err)
-					continue
-				}
-				w.checkUnavailble(ss, alert)
-
-			} else if alert.Spec.TargetWorkload.Selector != nil {
-
-				selector := labels.NewSelector()
-				for key, value := range alert.Spec.TargetWorkload.Selector {
-					r, err := labels.NewRequirement(key, selection.Equals, []string{value})
-					if err != nil {
-						logrus.Warnf("Fail to create new requirement foo %s: %v", key, err)
-						continue
-					}
-					selector = selector.Add(*r)
-				}
-				ss, err := w.statefulsetLister.List("", selector)
+
+		if alert.Spec.TargetWorkload.WorkloadID != "" {
+			parts := strings.Split(alert.Spec.TargetWorkload.WorkloadID, ":")
+			if len(parts) != 3 {
+				logrus.Warnf("Invalid workload ID format %s", alert.Spec.TargetWorkload.WorkloadID)
+				continue
+			}
+
+			workloadType := parts[0]
+			if workloadType != "statefulset" {
+				continue
+			}
+			ns := parts[1]
+			id := parts[2]
+			ss, err := w.statefulsetLister.Get(ns, id)
```

Comment:
you should utilize workload_common.go for listing the workload. Doing one function for type is too much work. https://github.com/rancher/rancher/blob/master/pkg/controllers/user/workload/workload_common.go

Reply:
Sure,  I will change to use workload_common and remove statefulset.go/daemonset.go/deployment,go